src/libcommon/log/sentry/handler.cpp

```diff
@@ -434,7 +446,158 @@ SentryHandler::~SentryHandler() {
     }
 }
 
-SentryHandler::SentryEvent::SentryEvent(const std::string &title, const std::string &message, SentryLevel level,
+Handler::SentryEvent::SentryEvent(const std::string &title, const std::string &message, Level level,
                                         SentryConfidentialityLevel confidentialityLevel, const SentryUser &user) :
-    title(title), message(message), level(level), confidentialityLevel(confidentialityLevel), userId(user.userId()) {}
-} // namespace KDC
+    title(title),
+    message(message), level(level), confidentialityLevel(confidentialityLevel), userId(user.userId()) {}
+
+void Handler::stopPTrace(const pTraceId &id, bool aborted) {
+    std::scoped_lock lock(_mutex);
+    if (id == 0) return;
```

Comment:
When does it happen? Can this be placed before we lock the mutex?

Reply:
If `Handler::startPTrace` fails, it will return 0 as the `pTraceId`.  
For the sake of performance and readability, the caller is not required to check this value. It can be used as is, and the `Handler` functions dealing with `pTraceId` will simply ignore any calls referring to an ID of 0.

Reply:
Moved here: [dc43287](https://github.com/Infomaniak/desktop-kDrive/pull/403/commits/dc432879d2bb5f405a700f650f96ff037efa30f2)